File: Daily/1523.count-odd-numbers-in-an-interval-range.js
/*
 * @lc app=leetcode id=1523 lang=javascript
 *
 * [1523] Count Odd Numbers in an Interval Range
 */

// @lc code=start
/**
 * @param {number} low
 * @param {number} high
 * @return {number}
 */
var countOdds = function (low, high) {
  if (isEven(low)) {
    low++;
  }
  if (isEven(high)) {
    high--;
  }

  return (high - low) / 2 + 1;
};

/**
 * @param {number} number
 * @returns {boolean}
 */
function isEven(number) {
  return (number & 0b1) === 0;
}
// @lc code=end

File: src/2025/1523.count-odd-numbers-in-an-interval-range.cpp
/*
 * @lc app=leetcode id=1523 lang=cpp
 *
 * [1523] Count Odd Numbers in an Interval Range
 */

// @lc code=start
class Solution {
 public:
  int countOdds(int low, int high) {
    int count = 0;
    if (low % 2 == 1) {
      count++;
      low++;
    }
    if (high % 2 == 1) {
      count++;
      high--;
    }

    count += (high - low) / 2;
    return count;
  }
};
// @lc code=end
